### ISSUE TITLE
Property 'name' is missing

### ISSUE THREAD

**Issue by hun1ahpu** [Microsoft/TypeScript-React-Starter#37]
After completing all the steps of the tutorial having an error at index.tsx: 19

`Property 'name' is missing in type 'IntrinsicAttributes & IntrinsicClassAttributes<Component<Props, ComponentState>> & Readonly<{ chi...'.	my-app (tsconfig project)`

Setting the name property fixes the issue:
`<Hello name="something" />`

**Comment by Excommunicated** [comment]
Adding a `?` after the name property declaration in the `Props` interface also fixes the issue.

```typescript
export interface Props {
    name?: string;
    enthusiasmLevel?: number;
    onIncrement?: () => void;
    onDecrement?: () => void;
}
```

**Comment by bonebizz21** [comment]
thanks

**Comment by hun1ahpu** [comment]
@Excommunicated i don't know if making 'name' property optional is correct fix here. 
Is it working at all for anyone without optional name?

**Comment by bonebizz21** [comment]
Umm.   Username possibly.   <Bonebizz21>

**Comment by Excommunicated** [comment]
@hun1ahpu I completely agree. I was just offering a solution that will work with the compiler as it stands, without having to set the property on the element.s
The underlying issue still remains. The compiler is not recognizing the `mapStateToProps` function mapping. If anything, it should be reporting if you were to omit setting `languageName` on the stores initial state.

**Comment by bzums** [comment]
Seems to be related to https://github.com/Microsoft/TypeScript-React-Starter/issues/29 right? Is the proposed solution there the way to go? And can somebody explain *why* exactly that works? Best

**Comment by VSDekar** [comment]
@DanielRosenwasser could you help? What are we missing? Since it should be possible to omit the 'name' property.

**Comment by VSDekar** [comment]
After some research, i think this bug is related to some changes in the typings for react-redux.
Stick the version in package.json to `@types/react-redux: "4.4.36"` (up to 4.4.40) should "Fix" this issue until a real bugfix is released.

**Comment by bonebizz21** [comment]
Thank u

**Comment by tebeco** [comment]
As said :

Seems more logic to let the property mandatory in the props
and change App.tsx from:
`<Hello />`

to `<Hello name="Whatever you want" />`

**Comment by DanielRosenwasser** [comment]
I'll try to look into this next week. Sorry!

**Comment by Dev-Dipesh** [comment]
For me it's simple typo issue, I forgot to update import statement for **Hello** component.

**Before** (*source of error*)
```js
import Hello from './components/Hello';
```

**After** (*fixed the issue*)
```js
import Hello from './containers/Hello';
```

**Comment by provDavidLaBine** [comment]
@Dev-Dipesh I had the same issue and was stumped for a little while.  Thanks for the fix!  My feeling is that this is likely what most people are running into, since it is oh so very subtly wrong.  I'd try this before any of the other solutions proposed.

**Comment by bonebizz21** [comment]
Layla G

**Comment by bonebizz21** [comment]
<Hello Sonia/>

**Comment by nedzelskiy** [comment]
I have this issue with dispatched property to connected component props. I get an error: Property 'property' is missing in type {} but required in type 'Readonly<Pick<Props, 'property'>> 
Can someone help me to avoid this error?
